server/services/survey.service.js

```diff
@@ -0,0 +1,60 @@
+const Path = require('path-parser');
+const { URL } = require('url');
+
+module.exports = class SurveyService {
+
+  async updateSurveys(data) {
+    try {
+
+      const uniqueServeys =
+          data.map(this.buildSurveyDataFromBody)
+              .filter(this.filterOutNullAndUndefined)
+              .filter(this.filterByUniqueEmailAndSurveyId);
+
+      for (let survey of uniqueServeys) {
+        await Survey.updateOne(
+            {
+              _id: survey.surveyId,
+              recipients: {
+                $elemMatch: {email: survey.email, responded: false}
+              }
+            },
+            {
+              $inc: {[survey.choice]: 1},
+              $set: {'recipients.$.responded': true},
+              lastResponded: new Date()
+            }
+        ).exec();
+      }
+
+      return { success: true }
+    } catch(error) {
+      return error;
+    }
+  }
+
+  getRecipientListFromEmailString(recipients) {
+    return recipients.split(',').map(email => {
+      return { email };
+    })
+  }
```

Comment:
This is that helper function for the recipient formatting, pretty basic, make an array from a comma separated string and then turn those strings into object with email as the key and the string as the value.